group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -1133,14 +1148,19 @@ public void testStaticMemberGetsBackAssignmentUponRejoin() {
             .build();
 
         List<CoordinatorRecord> expectedRecordsAfterRejoin = Arrays.asList(
+            // The previous member is deleted.
             CoordinatorRecordHelpers.newCurrentAssignmentTombstoneRecord(groupId, memberId2),
             CoordinatorRecordHelpers.newTargetAssignmentTombstoneRecord(groupId, memberId2),
             CoordinatorRecordHelpers.newMemberSubscriptionTombstoneRecord(groupId, memberId2),
-            CoordinatorRecordHelpers.newMemberSubscriptionRecord(groupId, expectedRejoinedMember),
+
+            // The previous member is replaced by the new one.
```

Comment:
I don't think I understand this comment. These are the records that get generated by `replaceMember`?

And so I understand the only difference between copy and rejoin is the member epoch?

Reply:
Correct. These are the records generated by `replaceMember`. There are a copy of the previous member but with a new member id, and new member epoch and a new previous member epoch.